File: recovered-source/jadx/sources/io/legado/app/utils/StringExtensionsKt.java
package io.legado.app.utils;

import io.legado.app.constant.AppPattern;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collection;
import java.util.List;
import kotlin.Metadata;
import kotlin.collections.CollectionsKt;
import kotlin.jvm.internal.Intrinsics;
import kotlin.text.Regex;
import kotlin.text.StringsKt;
import me.ag2s.epublib.epub.PackageDocumentBase;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/* JADX INFO: compiled from: StringExtensions.kt */
/* JADX INFO: loaded from: app-classes.jar:io/legado/app/utils/StringExtensionsKt.class */
@Metadata(mv = {1, 5, 1}, k = 2, xi = 48, d1 = {"\u0000&\n\u0000\n\u0002\u0010\b\n\u0002\u0010\u000e\n\u0002\b\u0003\n\u0002\u0010\u000b\n\u0002\b\t\n\u0002\u0010\u0011\n\u0002\b\u0003\n\u0002\u0018\u0002\n\u0002\b\u0007\u001a\u0012\u0010\u0000\u001a\u00020\u0001*\u00020\u00022\u0006\u0010\u0003\u001a\u00020\u0002\u001a\f\u0010\u0004\u001a\u00020\u0002*\u0004\u0018\u00010\u0002\u001a\f\u0010\u0005\u001a\u00020\u0006*\u0004\u0018\u00010\u0002\u001a\f\u0010\u0007\u001a\u00020\u0006*\u0004\u0018\u00010\u0002\u001a\f\u0010\b\u001a\u00020\u0006*\u0004\u0018\u00010\u0002\u001a\f\u0010\t\u001a\u00020\u0006*\u0004\u0018\u00010\u0002\u001a\f\u0010\n\u001a\u00020\u0006*\u0004\u0018\u00010\u0002\u001a\u0016\u0010\u000b\u001a\u00020\u0006*\u0004\u0018\u00010\u00022\b\b\u0002\u0010\f\u001a\u00020\u0006\u001a\f\u0010\r\u001a\u00020\u0006*\u0004\u0018\u00010\u0002\u001a\u000e\u0010\u000e\u001a\u0004\u0018\u00010\u0002*\u0004\u0018\u00010\u0002\u001a)\u0010\u000f\u001a\b\u0012\u0004\u0012\u00020\u00020\u0010*\u00020\u00022\u0012\u0010\u0011\u001a\n\u0012\u0006\b\u0001\u0012\u00020\u00020\u0010\"\u00020\u0002¢\u0006\u0002\u0010\u0012\u001a'\u0010\u000f\u001a\b\u0012\u0004\u0012\u00020\u00020\u0010*\u00020\u00022\u0006\u0010\u0013\u001a\u00020\u00142\b\b\u0002\u0010\u0015\u001a\u00020\u0001¢\u0006\u0002\u0010\u0016\u001a\u0012\u0010\u0017\u001a\u00020\u0006*\u00020\u00022\u0006\u0010\u0018\u001a\u00020\u0002\u001a\u0015\u0010\u0019\u001a\b\u0012\u0004\u0012\u00020\u00020\u0010*\u00020\u0002¢\u0006\u0002\u0010\u001a¨\u0006\u001b"}, d2 = {"cnCompare", PackageDocumentBase.PREFIX_OPF, PackageDocumentBase.PREFIX_OPF, "other", "htmlFormat", "isAbsUrl", PackageDocumentBase.PREFIX_OPF, "isDataUrl", "isJson", "isJsonArray", "isJsonObject", "isTrue", "nullIsTrue", "isXml", "safeTrim", "splitNotBlank", PackageDocumentBase.PREFIX_OPF, "delimiter", "(Ljava/lang/String;[Ljava/lang/String;)[Ljava/lang/String;", "regex", "Lkotlin/text/Regex;", "limit", "(Ljava/lang/String;Lkotlin/text/Regex;I)[Ljava/lang/String;", "startWithIgnoreCase", "start", "toStringArray", "(Ljava/lang/String;)[Ljava/lang/String;", "reader-pro"})
public final class StringExtensionsKt {
    @Nullable
    public static final String safeTrim(@Nullable String $this$safeTrim) {
        String str = $this$safeTrim;
        if (str == null || StringsKt.isBlank(str)) {
            return null;
        }
        if ($this$safeTrim == null) {
            throw new NullPointerException("null cannot be cast to non-null type kotlin.CharSequence");
        }
        return StringsKt.trim($this$safeTrim).toString();
    }

    public static final boolean isAbsUrl(@Nullable String $this$isAbsUrl) {
        String str = $this$isAbsUrl;
        if (str == null || StringsKt.isBlank(str)) {
            return false;
        }
        return StringsKt.startsWith($this$isAbsUrl, "http://", true) || StringsKt.startsWith($this$isAbsUrl, "https://", true);
    }

    public static final boolean isDataUrl(@Nullable String $this$isDataUrl) {
        if ($this$isDataUrl == null) {
            return false;
        }
        return AppPattern.INSTANCE.getDataUriRegex().matches($this$isDataUrl);
    }

    public static final boolean isJson(@Nullable String $this$isJson) {
        if ($this$isJson == null) {
            return false;
        }
        String str = StringsKt.trim($this$isJson).toString();
        return (StringsKt.startsWith$default(str, "{", false, 2, (Object) null) && StringsKt.endsWith$default(str, "}", false, 2, (Object) null)) ? true : StringsKt.startsWith$default(str, "[", false, 2, (Object) null) && StringsKt.endsWith$default(str, "]", false, 2, (Object) null);
    }

    public static final boolean isJsonObject(@Nullable String $this$isJsonObject) {
        if ($this$isJsonObject == null) {
            return false;
        }
        String str = StringsKt.trim($this$isJsonObject).toString();
        return StringsKt.startsWith$default(str, "{", false, 2, (Object) null) && StringsKt.endsWith$default(str, "}", false, 2, (Object) null);
    }

    public static final boolean isJsonArray(@Nullable String $this$isJsonArray) {
        if ($this$isJsonArray == null) {
            return false;
        }
        String str = StringsKt.trim($this$isJsonArray).toString();
        return StringsKt.startsWith$default(str, "[", false, 2, (Object) null) && StringsKt.endsWith$default(str, "]", false, 2, (Object) null);
    }

    public static final boolean isXml(@Nullable String $this$isXml) {
        if ($this$isXml == null) {
            return false;
        }
        String str = StringsKt.trim($this$isXml).toString();
        return StringsKt.startsWith$default(str, "<", false, 2, (Object) null) && StringsKt.endsWith$default(str, ">", false, 2, (Object) null);
    }

    public static /* synthetic */ boolean isTrue$default(String str, boolean z, int i, Object obj) {
        if ((i & 1) != 0) {
            z = false;
        }
        return isTrue(str, z);
    }

    public static final boolean isTrue(@Nullable String $this$isTrue, boolean nullIsTrue) {
        String str = $this$isTrue;
        if ((str == null || StringsKt.isBlank(str)) || Intrinsics.areEqual($this$isTrue, "null")) {
            return nullIsTrue;
        }
        return !new Regex("\\s*(?i)(false|no|not|0)\\s*").matches($this$isTrue);
    }

    @NotNull
    public static final String htmlFormat(@Nullable String $this$htmlFormat) {
        String str = $this$htmlFormat;
        if (str == null || StringsKt.isBlank(str)) {
            return PackageDocumentBase.PREFIX_OPF;
        }
        return new Regex("[\\n\\s]+$").replace(new Regex("^[\\n\\s]+").replace(new Regex("\\s*\\n+\\s*").replace(new Regex("<[script>]*.*?>|&nbsp;").replace(new Regex("(?i)<(br[\\s/]*|/*p\\b.*?|/*div\\b.*?)>").replace($this$htmlFormat, "\n"), PackageDocumentBase.PREFIX_OPF), "\n\u3000\u3000"), "\u3000\u3000"), PackageDocumentBase.PREFIX_OPF);
    }

    @NotNull
    public static final String[] splitNotBlank(@NotNull String $this$splitNotBlank, @NotNull String... delimiter) {
        Intrinsics.checkNotNullParameter($this$splitNotBlank, "<this>");
        Intrinsics.checkNotNullParameter(delimiter, "delimiter");
        Iterable $this$map$iv = StringsKt.split$default($this$splitNotBlank, (String[]) Arrays.copyOf(delimiter, delimiter.length), false, 0, 6, (Object) null);
        Collection destination$iv$iv = new ArrayList(CollectionsKt.collectionSizeOrDefault($this$map$iv, 10));
        for (Object item$iv$iv : $this$map$iv) {
            String it = (String) item$iv$iv;
            if (it == null) {
                throw new NullPointerException("null cannot be cast to non-null type kotlin.CharSequence");
            }
            destination$iv$iv.add(StringsKt.trim(it).toString());
        }
        Iterable $this$filterNot$iv = (List) destination$iv$iv;
        Collection destination$iv$iv2 = new ArrayList();
        for (Object element$iv$iv : $this$filterNot$iv) {
            if (!StringsKt.isBlank((String) element$iv$iv)) {
                destination$iv$iv2.add(element$iv$iv);
            }
        }
        Collection $this$toTypedArray$iv = (List) destination$iv$iv2;
        Object[] array = $this$toTypedArray$iv.toArray(new String[0]);
        if (array == null) {
            throw new NullPointerException("null cannot be cast to non-null type kotlin.Array<T>");
        }
        return (String[]) array;
    }

    @NotNull
    public static final String[] splitNotBlank(@NotNull String $this$splitNotBlank, @NotNull Regex regex, int limit) {
        Intrinsics.checkNotNullParameter($this$splitNotBlank, "<this>");
        Intrinsics.checkNotNullParameter(regex, "regex");
        Iterable $this$map$iv = regex.split($this$splitNotBlank, limit);
        Collection destination$iv$iv = new ArrayList(CollectionsKt.collectionSizeOrDefault($this$map$iv, 10));
        for (Object item$iv$iv : $this$map$iv) {
            String it = (String) item$iv$iv;
            if (it == null) {
                throw new NullPointerException("null cannot be cast to non-null type kotlin.CharSequence");
            }
            destination$iv$iv.add(StringsKt.trim(it).toString());
        }
        Iterable $this$filterNot$iv = (List) destination$iv$iv;
        Collection destination$iv$iv2 = new ArrayList();
        for (Object element$iv$iv : $this$filterNot$iv) {
            if (!StringsKt.isBlank((String) element$iv$iv)) {
                destination$iv$iv2.add(element$iv$iv);
            }
        }
        Collection $this$toTypedArray$iv = (List) destination$iv$iv2;
        Object[] array = $this$toTypedArray$iv.toArray(new String[0]);
        if (array == null) {
            throw new NullPointerException("null cannot be cast to non-null type kotlin.Array<T>");
        }
        return (String[]) array;
    }

    public static /* synthetic */ String[] splitNotBlank$default(String str, Regex regex, int i, int i2, Object obj) {
        if ((i2 & 2) != 0) {
            i = 0;
        }
        return splitNotBlank(str, regex, i);
    }

    public static final boolean startWithIgnoreCase(@NotNull String $this$startWithIgnoreCase, @NotNull String start) {
        Intrinsics.checkNotNullParameter($this$startWithIgnoreCase, "<this>");
        Intrinsics.checkNotNullParameter(start, "start");
        if (StringsKt.isBlank($this$startWithIgnoreCase)) {
            return false;
        }
        return StringsKt.startsWith($this$startWithIgnoreCase, start, true);
    }

    public static final int cnCompare(@NotNull String $this$cnCompare, @NotNull String other) {
        Intrinsics.checkNotNullParameter($this$cnCompare, "<this>");
        Intrinsics.checkNotNullParameter(other, "other");
        return $this$cnCompare.compareTo(other);
    }

    @NotNull
    public static final String[] toStringArray(@NotNull String $this$toStringArray) {
        String[] strArr;
        Intrinsics.checkNotNullParameter($this$toStringArray, "<this>");
        int codePointIndex = 0;
        try {
            int iCodePointCount = $this$toStringArray.codePointCount(0, $this$toStringArray.length());
            String[] strArr2 = new String[iCodePointCount];
            for (int i = 0; i < iCodePointCount; i++) {
                int start = codePointIndex;
                codePointIndex = $this$toStringArray.offsetByCodePoints(start, 1);
                String strSubstring = $this$toStringArray.substring(start, codePointIndex);
                Intrinsics.checkNotNullExpressionValue(strSubstring, "(this as java.lang.Strin…ing(startIndex, endIndex)");
                strArr2[i] = strSubstring;
            }
            strArr = strArr2;
        } catch (Exception e) {
            Collection $this$toTypedArray$iv = StringsKt.split$default($this$toStringArray, new String[]{PackageDocumentBase.PREFIX_OPF}, false, 0, 6, (Object) null);
            Object[] array = $this$toTypedArray$iv.toArray(new String[0]);
            if (array == null) {
                throw new NullPointerException("null cannot be cast to non-null type kotlin.Array<T>");
            }
            strArr = (String[]) array;
        }
        return strArr;
    }
}

File: best-of-3/src/io/legado/app/utils/StringExtensionsKt.kt
package io.legado.app.utils

import io.legado.app.constant.AppPattern
import java.util.ArrayList
import java.util.Arrays

public fun String?.safeTrim(): String? {
   val var10000: java.lang.String;
   if (`$this$safeTrim` == null || StringsKt.isBlank(`$this$safeTrim`)) {
      var10000 = null;
   } else {
      if (`$this$safeTrim` == null) {
         throw new NullPointerException("null cannot be cast to non-null type kotlin.CharSequence");
      }

      var10000 = StringsKt.trim(`$this$safeTrim`).toString();
   }

   return var10000;
}

public fun String?.isAbsUrl(): Boolean {
   return `$this$isAbsUrl` != null
      && !StringsKt.isBlank(`$this$isAbsUrl`)
      && (StringsKt.startsWith(`$this$isAbsUrl`, "http://", true) || StringsKt.startsWith(`$this$isAbsUrl`, "https://", true));
}

public fun String?.isDataUrl(): Boolean {
   return `$this$isDataUrl` != null && AppPattern.INSTANCE.getDataUriRegex().matches(`$this$isDataUrl`);
}

public fun String?.isJson(): Boolean {
   val var10000: Boolean;
   if (`$this$isJson` == null) {
      var10000 = false;
   } else {
      val str: java.lang.String = StringsKt.trim(`$this$isJson`).toString();
      var10000 = StringsKt.startsWith$default(str, "{", false, 2, null) && StringsKt.endsWith$default(str, "}", false, 2, null)
         || StringsKt.startsWith$default(str, "[", false, 2, null) && StringsKt.endsWith$default(str, "]", false, 2, null);
   }

   return var10000;
}

public fun String?.isJsonObject(): Boolean {
   val var10000: Boolean;
   if (`$this$isJsonObject` == null) {
      var10000 = false;
   } else {
      val str: java.lang.String = StringsKt.trim(`$this$isJsonObject`).toString();
      var10000 = StringsKt.startsWith$default(str, "{", false, 2, null) && StringsKt.endsWith$default(str, "}", false, 2, null);
   }

   return var10000;
}

public fun String?.isJsonArray(): Boolean {
   val var10000: Boolean;
   if (`$this$isJsonArray` == null) {
      var10000 = false;
   } else {
      val str: java.lang.String = StringsKt.trim(`$this$isJsonArray`).toString();
      var10000 = StringsKt.startsWith$default(str, "[", false, 2, null) && StringsKt.endsWith$default(str, "]", false, 2, null);
   }

   return var10000;
}

public fun String?.isXml(): Boolean {
   val var10000: Boolean;
   if (`$this$isXml` == null) {
      var10000 = false;
   } else {
      val str: java.lang.String = StringsKt.trim(`$this$isXml`).toString();
      var10000 = StringsKt.startsWith$default(str, "<", false, 2, null) && StringsKt.endsWith$default(str, ">", false, 2, null);
   }

   return var10000;
}

public fun String?.isTrue(nullIsTrue: Boolean = false): Boolean {
   if (`$this$isTrue` != null && !StringsKt.isBlank(`$this$isTrue`) && !(`$this$isTrue` == "null")) {
      return !new Regex("\\s*(?i)(false|no|not|0)\\s*").matches(`$this$isTrue`);
   } else {
      return nullIsTrue;
   }
}

@JvmSynthetic
fun `isTrue$default`(var0: java.lang.String, var1: Boolean, var2: Int, var3: Any): Boolean {
   if ((var2 and 1) != 0) {
      var1 = false;
   }

   return isTrue(var0, var1);
}

public fun String?.htmlFormat(): String {
   return if (`$this$htmlFormat` as java.lang.CharSequence == null || StringsKt.isBlank(`$this$htmlFormat`))
      ""
      else
      new Regex("[\\n\\s]+$")
         .replace(
            new Regex("^[\\n\\s]+")
               .replace(
                  new Regex("\\s*\\n+\\s*")
                     .replace(
                        new Regex("<[script>]*.*?>|&nbsp;").replace(new Regex("(?i)<(br[\\s/]*|/*p\\b.*?|/*div\\b.*?)>").replace(`$this$htmlFormat`, "\n"), ""),
                        "\n　　"
                     ),
                  "　　"
               ),
            ""
         );
}

public fun String.splitNotBlank(vararg delimiter: String): Array<String> {
   var `$this$toTypedArray$iv`: java.lang.Iterable = StringsKt.split$default(
      `$this$splitNotBlank`, Arrays.copyOf(delimiter, delimiter.length), false, 0, 6, null
   );
   var `destination$iv$iv`: java.util.Collection = new ArrayList(CollectionsKt.collectionSizeOrDefault(`$this$toTypedArray$iv`, 10));

   for (Object item$iv$iv : $this$map$iv) {
      val it: java.lang.String = `element$iv$iv` as java.lang.String;
      if (`element$iv$iv` as java.lang.String == null) {
         throw new NullPointerException("null cannot be cast to non-null type kotlin.CharSequence");
      }

      `destination$iv$iv`.add(StringsKt.trim(it).toString());
   }

   `$this$toTypedArray$iv` = `destination$iv$iv` as java.util.List;
   `destination$iv$iv` = new ArrayList();

   for (Object element$iv$iv : $this$map$iv) {
      if (!StringsKt.isBlank(var27 as java.lang.String)) {
         `destination$iv$iv`.add(var27);
      }
   }

   val var10000: Array<Any> = (`destination$iv$iv` as java.util.List).toArray(new java.lang.String[0]);
   if (var10000 == null) {
      throw new NullPointerException("null cannot be cast to non-null type kotlin.Array<T>");
   } else {
      return var10000 as Array<java.lang.String>;
   }
}

public fun String.splitNotBlank(regex: Regex, limit: Int = 0): Array<String> {
   var var21: java.lang.Iterable = regex.split(`$this$splitNotBlank`, limit);
   var `destination$iv$iv`: java.util.Collection = new ArrayList(CollectionsKt.collectionSizeOrDefault(var21, 10));

   for (Object item$iv$iv : var21) {
      val it: java.lang.String = `element$iv$iv` as java.lang.String;
      if (`element$iv$iv` as java.lang.String == null) {
         throw new NullPointerException("null cannot be cast to non-null type kotlin.CharSequence");
      }

      `destination$iv$iv`.add(StringsKt.trim(it).toString());
   }

   var21 = `destination$iv$iv` as java.util.List;
   `destination$iv$iv` = new ArrayList();

   for (Object element$iv$iv : var21) {
      if (!StringsKt.isBlank(var30 as java.lang.String)) {
         `destination$iv$iv`.add(var30);
      }
   }

   val var10000: Array<Any> = (`destination$iv$iv` as java.util.List).toArray(new java.lang.String[0]);
   if (var10000 == null) {
      throw new NullPointerException("null cannot be cast to non-null type kotlin.Array<T>");
   } else {
      return var10000 as Array<java.lang.String>;
   }
}

@JvmSynthetic
fun `splitNotBlank$default`(var0: java.lang.String, var1: Regex, var2: Int, var3: Int, var4: Any): Array<java.lang.String> {
   if ((var3 and 2) != 0) {
      var2 = 0;
   }

   return splitNotBlank(var0, var1, var2);
}

public fun String.startWithIgnoreCase(start: String): Boolean {
   return !StringsKt.isBlank(`$this$startWithIgnoreCase`) && StringsKt.startsWith(`$this$startWithIgnoreCase`, start, true);
}

public fun String.cnCompare(other: String): Int {
   return `$this$cnCompare`.compareTo(other);
}

public fun String.toStringArray(): Array<String> {
   var codePointIndex: Int = 0;

   var var2: Array<java.lang.String>;
   try {
      var var11: Int = 0;
      val e: Int = `$this$toStringArray`.codePointCount(0, `$this$toStringArray`.length());

      val var13: Array<java.lang.String>;
      for (var13 = new java.lang.String[e]; var11 < e; var11++) {
         val var15: Int = codePointIndex;
         codePointIndex = `$this$toStringArray`.offsetByCodePoints(codePointIndex, 1);
         val var10002: java.lang.String = `$this$toStringArray`.substring(var15, codePointIndex);
         var13[var11] = var10002;
      }

      var2 = var13;
   } catch (var10: Exception) {
      val var17: Array<Any> = StringsKt.split$default(`$this$toStringArray`, new java.lang.String[]{""}, false, 0, 6, null).toArray(new java.lang.String[0]);
      if (var17 == null) {
         throw new NullPointerException("null cannot be cast to non-null type kotlin.Array<T>");
      }

      var2 = var17 as Array<java.lang.String>;
   }

   return var2;
}
